dos/src/main.rs

```diff
@@ -279,8 +308,18 @@ fn run_dos(
 
 // command line parsing
 #[derive(Parser)]
-#[clap(name = crate_name!(), version = crate_version!(), about = crate_description!())]
+#[clap(name = crate_name!(),
+    version = crate_version!(),
+    about = crate_description!(),
+    rename_all = "kebab-case"
+)]
 struct DosClientParameters {
+    #[clap(long, arg_enum, help = "Interface to DoS")]
+    mode: Mode,
+
+    #[clap(long, arg_enum, help = "Type of data to send")]
+    data_type: DataType,
```

Comment:
🎉

Reply:
yeah, this clap's `arg_enum` was a missing piece, now I'm really impressed by clap v3 and it's Parser.